src/MMM01.md

```diff
@@ -0,0 +1,190 @@
+
+External RAM is often battery-backed, allowing for the storage of game data while the Game Boy is turned off, or if the cartridge is removed from the Game Boy.
+
+It is currently unknown whether RAM access is possible while in unmapped mode.
+
+## Registers
+
+The MMM01 registers are extensions of the MBC1 registers. In "mapped" mode, the extra bits are unwriteable and the MMM01 emulates an MBC1. In "unmapped" mode, the extra bits are writeable, allowing the menu program to select which game to play and configure its size and RAM access.
+
+All the MMM01 registers are 7-bits in size and set to 00h on power-up. For the ROM Bank Number register, this behaves as if it was set to 01h.
+
+### 0000-1FFF - RAM Enable (Write Only)
+
+```
+Bits: X 6 5 4 3 2 1 0
+        | \_/ \_____/
+        |  |      \----- Bits 0-3: RAM Enable
+        |  \------------ Bits 4-5: RAM Bank Mask
+        \--------------- Bit 6: Mapping Enable
+```
+
+#### Bits 0-3: RAM Enable
+
+As per MBC1, writing 0Ah here enables the RAM, and any other value disables the RAM. RAM is automatically disabled when the gameboy is powered off.
+
+#### Bits 4-5: RAM Bank Mask
+
+Can only be written in unmapped mode.
+
+If either or both of these bits are set high, the corresponding bit(s) in the MBC1 RAM banking register are locked in as "game select" bits instead of "banking" bits. Any attempt to write to those bits while its matching bit in this register is 1 is prevented.
+
+#### Bit 6: Mapping Enable
+
+Can only be written in unmapped mode.
+
+If 1, the MMM01 enters "mapped" mode. This immediately begins MBC1 emulation, mapping in the selected banks of the game ROM (typically 00h and 01h within the game's subsection of the cartridge ROM) and prevents write access to any of the MMM01 extended control bits.
+
+It is unknown if setting bit 6 to enter mapping mode will prevent simultaneous writes to bits 4-5 (RAM Bank Mask). The only released MMM01 cartridge containing RAM performs two separate writes to set the bank mask before enabling mapping.
+
+### 2000-3FFF - ROM Bank Number (Write Only)
+
+```
+Bits: X 6 5 4 3 2 1 0
+        \_/ \_______/
+         |       \------ Bits 0-4: ROM Bank Low
+         \-------------- Bits 5-6: ROM Bank Mid
+```
+
+#### Bits 0-4: ROM Bank Low
+
+This is equivalent to the MBC1 ROM Bank register, and (when in mapped mode, in addition to the ROM Bank Mid/High game select registers) selects the ROM bank visible in the 4000-7FFF region. It can be masked to reduce its size, reserving some of the bits for game select (see "ROM Bank Mask"). Bits masked for game select also affect the bank visible in the 0000-3FFF region.
+
+If the _unmasked_ bits are all 0, it behaves as if the lowest bit is set to 1 (as per MBC1 behaviour, mapping bank 00 maps bank 01). However, the actual value of the register doesn't change.
```

Comment:
I've attempted to explain it in the mask section, but here's an example:

$10 written to ROM Bank Low
$10 written to ROM Bank Mask
// entering mapped mode at this point would map actual bank $11 to 4000-7FFF because the unmasked bits are all 0s

So unlike MBC1, the remapping isn't only triggered by a write of $00 to the ROM Bank register. I believe MBC1 is the same (only using a 1, but storing it as 0) but the difference is not observable on an actual MBC1.

Reply:
Ok, yeah, it's observable, so definitely backed up. Could this rationale be noted? For example:
```suggestion
If all _unmasked_ bits equal 0, the lowest bit, which is always writable, behaves as if set to 1 (as per MBC1 behaviour, mapping bank $00 maps bank $01).
However, the actual value of the register doesn't change, as evidenced if changing the [masking register](...) to un-trigger this behavior.
```